src/shogun/base/SGObject.h

```diff
@@ -65,6 +65,11 @@ bool dispatch_array_type(const CSGObject* obj, std::string_view name,
 #endif // DOXYGEN_SHOULD_SKIP_THIS
 #endif // SWIG
 
+struct InterfaceVisitor
```

Comment:
this might be better to place into a separate .h file and include it here. place it under src/shogun/util/visitors